client/client_role_test.go

```diff
@@ -62,6 +65,26 @@ func NewSetups(rng *rand.Rand, names []string) []ctest.RoleSetup {
 	return setup
 }
 
+type Client struct {
+	*client.Client
+	ctest.RoleSetup
+}
```

Comment:
I know that we are using the roles in the other tests, but this has downsides.

1. It makes the tests hard to understand. I also had a lot of trouble in the past working with the roles because of this and debugging was difficult.
2. I don't believe the same roles work for the virtual channel tests, even though I didn't check in detail, and writing similar roles is just a lot of work and I don't see the benefits.

Besides that, I already simplified the tests in the following commits that are left. You can have a look at #83 .